### DISCUSSION TITLE
Omit if keyword after else expression

### DISCUSSION THREAD

**Discussion by dmitryuck** [vlang/v#18691] [Ideas]
### Describe the feature

The suggestion is to make `if` keyword not required after `else`

if a > b { a } else a == b { b } else { a + b }

### Use Case

I am always frustrating when I write if after else

if a < b {
	println('${a} < ${b}')
} else a > b {
	println('${a} > ${b}')
}

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### Version used

V 0.3.4 046dd54

### Environment details (OS name and version, etc.)

Windows,
V 0.3.4 046dd54

**Comment by JalonSolov** [comment]
I don't see this one happening.  It is both a LARGE breaking change to existing code, as well as a departure from all other languages, just to save 3 extra characters.

**Comment by ttytm** [comment]
The extra chars from `if` also serve as semantical indicator that it's another condition that's coming. And it does it before one needs to read the upcoming conditional expression, just by keywords. So having it makes it easier and in terms of knowing the scheme of whats going on also **less** to read. `else` without if is just all other cases - so also imho, it should stay that way.

**Comment by dmitryuck** [comment]
> It is both a LARGE breaking change to existing code

In the first step, we can allow use if after else, just add compiler an ability to compile
after some time add a compiler message about deprecating, and finally change at 0.5 or 0.6 version

> The extra chars from if the also serve as an semantical indicator that it's another condition that's coming

we can give an additional meaning for else, that can be a conditions also, if an expression follows next. 

we build a new language easy and powerful, some other languages think they need to have a () for if, here we did as kotlin, rust  and go, we just skipped them as a redundant, why not to give people write less and read easier. Plus I see no bivalent meaning in this.

**Comment by Wajinn** [comment]
> if a > b { a } else a == b { b } else { a + b }

`if a > b { a } else if a == b { b } else { a + b }`

In terms of readability, a stream of `else` statements could confuse others, particularly those coming from other languages and are not starting from the beginning.  The conditional statement `else if`, and its other forms like `elif`, have a long history in computer languages.  As far as I'm aware of, Kotlin, Rust, and Go have `else if`.

Maybe a better argument would be to ask for replacing `else if` with `elif`.  The functionality and purpose is the same, but less words.  But, this late in development, seems questionable.  Surely many would question such a change on the grounds of clarity or tradition.

If a person doesn't want to deal with `else if`, then they can use [match](https://github.com/vlang/v/blob/master/doc/docs.md#match) (switch or case in other languages).  However, Vlang gives the convenience of being able to write an if-else if-else chain in one line (like how some languages use ternary operators).

**Comment by dmitryuck** [comment]
> Maybe a better argument would be to ask for replacing else if with elif

no, `elif` definitely no, it looks ugly.

**Comment by i582** [comment]
please no, no reason for this
